config/initializers/controllers.rb

```diff
@@ -1,98 +1,100 @@
-  # but gem controllers like FinePrint's and Blazer's do not, so adding them there wouldn't work
-  # However, all controllers inherit from ActionController::Base,
-  # so adding these methods to ActionController::Base directly here should work
-  include UserSessionManagement
-  include ApplicationHelper
-  include OSU::OsuHelper
-  include HttpAcceptLanguage::AutoLocale
-  include RequireRecentSignin
-
-  include AuthenticateMethods
-
-  include ContractsNotRequired
-  helper_method :contracts_not_required
-
-  helper OSU::OsuHelper, ApplicationHelper, UserSessionManagement
-
-  prepend_before_action :set_device_id
-  before_action :save_redirect
-  before_action :set_locale
-  before_action :complete_signup_profile
-
-  fine_print_require :general_terms_of_use, :privacy_policy, unless: :disable_fine_print
-
-  protected
-
-  def complete_signup_profile
-    return true if request.format != :html || request.options?
-    redirect_to '/signup/profile' if current_user.is_needs_profile?
-    # TODO: uncomment this line after fixing openstax_path_prefixer
-    # redirect_to main_app.signup_profile_path if current_user.is_needs_profile?
-  end
-
-  def security_log(event_type, event_data = {})
-    user = event_data[:user]
-
-    if respond_to?(:current_api_user)
-      api_user = current_api_user
-      user ||= api_user.human_user
-      application = api_user.application
-    else
-      user ||= current_user
-      application = nil
+ActiveSupport.on_load(:action_controller_base) do
+  ActionController::Base.class_exec do
+    # We need some of the methods added here in all controllers
+    # For example, `current_user` is used in the layout so it needs to exist everywhere
+    # OpenStax Accounts's controllers all inherit from ApplicationController,
+    # but gem controllers like FinePrint's and Blazer's do not, so adding them there wouldn't work
+    # However, all controllers inherit from ActionController::Base,
+    # so adding these methods to ActionController::Base directly here should work
+    include UserSessionManagement
+    include ApplicationHelper
+    include OSU::OsuHelper
+    include HttpAcceptLanguage::AutoLocale
+    include RequireRecentSignin
+
+    include AuthenticateMethods
+
+    include ContractsNotRequired
+    helper_method :contracts_not_required
+
+    helper OSU::OsuHelper, ApplicationHelper, UserSessionManagement
+
+    prepend_before_action :set_device_id
+    before_action :save_redirect
+    before_action :set_locale
+    before_action :complete_signup_profile
+
+    fine_print_require :general_terms_of_use, :privacy_policy, unless: :disable_fine_print
+
+    protected
+
+    def complete_signup_profile
+      return true if request.format != :html || request.options?
+      redirect_to '/signup/profile' if current_user.is_needs_profile?
+      # TODO: uncomment this line after fixing openstax_path_prefixer
+      # redirect_to main_app.signup_profile_path if current_user.is_needs_profile?
```

Comment:
is this still a problem?
actually, i don't even thing /signup/profile is a path - so maybe best to leave this can of worms unopened